File: DOHEE/Week1/신고 결과 받기.py
# 16 분
def solution(id_list, report, k):
    cake = {}
    people = {}
    for i in id_list:
        cake[i] = {"count": 0, "reporter": []}
        people[i] = 0
        
    for i in report:
        reporter, reported = i.split(" ")
        if reporter not in cake[reported]['reporter']:
            cake[reported]['count'] += 1
            cake[reported]['reporter'].append(reporter)
        
    for i in cake:
        if cake.get(i)['count'] >= k:
            reporters = cake.get(i)['reporter']
            for j in reporters:
                people[j] += 1

    answer = [people[i] for i in people]
    return answer

File: DOHEE/Week1/신고 결과 받기.js
// 17분
function solution(id_list, report, k) {
    const cake = {};
    const people = {};
    id_list.forEach((v) => {
        cake[v] = { count: 0, people: [] };
        people[v] = 0;
    });
    report.forEach((rpt) => {
        const [reporter, reported] = rpt.split(" ");
        if (cake[reported].people.indexOf(reporter) === -1) {
            cake[reported].count += 1;
            cake[reported].people.push(reporter);
        }
    });
    Object.values(cake).forEach((v) => {
        if (v.count >= k) v.people.forEach((u) => (people[u] += 1));
    });
    var answer = Object.values(people);
    return answer;
}
